shared/js/content-scripts/click-to-load.js

```diff
@@ -663,10 +670,11 @@
 
     function replaceClickToLoadElements (config) {
         for (const entity of Object.keys(config)) {
-            for (const widget of Object.values(config[entity].elementData)) {
-                const els = document.querySelectorAll(widget.selectors.join())
-                for (const el of els) {
-                    createReplacementWidget(entity, widget, el)
+            for (const widgetData of Object.values(config[entity].elementData)) {
+                const trackingElements = document.querySelectorAll(widgetData.selectors.join())
+                for (const trackingElement of trackingElements) {
+                    const widget = new DuckWidget(widgetData, trackingElement, entity)
+                    createPlaceholderElementAndReplace(widget, trackingElement)
```

Comment:
This is an async function, but you're not assigning the promise returned by it. I wonder if this could introduce a race condition if a subsequent call should await the placeholders being created?

Reply:
Yea, you're completely right. I debated `await`ing here, but decided against. I figured we don't need (or want) to perform these calls in series and we don't (yet!) need to wait until `replaceClickToLoadElements` has finished. With my next change I will make `replaceClickToLoadElements` wait for those calls  so that we can dispatch a `"ddg-ctp-ready"` event reliably.